src/main/groovy/com/justinfield/dependencygraphdemo/DependencyGrapher.groovy

```diff
@@ -72,15 +72,27 @@ class DependencyGrapher {
         deps.each { String dependency ->
             // I made and assumption that if we detect a loop that we can just print the looped dep and not recurse
             if (dependencies.containsKey(dependency) && ! visited.contains(dependency)) {
+                // lets keep track of the depths that dont have anymore nodes when we recurse
+                if (deps.last() == dependency) {
+                    dontPrintSet.add(depth)
```

Comment:
Keeping track of nodes that are last(), will allow us to know as we recurse what depth levels to have ' ' instead of '|'